IPython/utils/text.py

```diff
@@ -561,7 +561,13 @@ def _vformat(self, format_string, args, kwargs, used_args, recursion_depth):
 
                 # eval the contents of the field for the object
                 # to be formatted
-                obj = eval(field_name, kwargs)
+                try:
+                    obj = eval(field_name, kwargs)
+                except NameError:
+                    if supressExceptions:
+                       obj = field_name
```

Comment:
~~An interesting point is that it's currently impossible to have a bash-ish $HOME and a python-ish $myvar at the same time...~~ No, actually the bash-ish could be written $$HOME and all's fine (the dollar parser recognises $$ and doesn't eval, replacing it with a single $).

Certainly in my case, I'm much more likely to use python variables than bash variables, though I can imagine cases where it may be necessary? That said, the bash/environment variables could always be put in a %%bash cell that calls the script you want.

I guess I'm naturally biased in this, but imho the bizarreness of warrants a warning more than the inconvenience of warning on use of $HOME.